lib/rubygems/specification.rb

```diff
@@ -782,9 +782,15 @@ def self.default_stubs pattern
   private_class_method :default_stubs
 
   def self.installed_stubs dirs, pattern
-    map_stubs(dirs, pattern) do |path, base_dir, gems_dir|
+    t = map_stubs(dirs, pattern) do |path, base_dir, gems_dir|
       Gem::StubSpecification.gemspec_stub(path, base_dir, gems_dir)
     end
+    # if mingw, derive platform from file name (loaded_from)
+    !RUBY_PLATFORM.match('mingw') ? t :
```

Comment:
Please rewrite without the ternary operator.

Reply:
Changed & rebased